Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -582,59 +640,56 @@ open class ComposerVC: _ViewController,
             }
         }
         composerView.inputMessageView.attachmentsViewContainer.isHidden = content.attachments.isEmpty
+    }
 
+    open func updateLinkPreview() {
         // Since we don't want to show link previews with other attachment types, we dismiss the
         // link preview in case it is being shown and there are other types of attachments in the message.
         if content.hasOnlyLinkAttachments == false && content.skipEnrichUrl == false {
             dismissLinkPreview()
         }
+    }
 
+    open func updateCheckbox() {
         if content.isInsideThread {
             if channelController?.channel?.isDirectMessageChannel == true {
                 composerView.checkboxControl.label.text = L10n.Composer.Checkmark.directMessageReply
             } else {
                 composerView.checkboxControl.label.text = L10n.Composer.Checkmark.channelReply
             }
         }
+    }
+
+    open func updateBottomContainer() {
         Animate {
             self.composerView.bottomContainer.isHidden = !self.content.isInsideThread
         }
+    }
 
+    open func updateCommandSuggestions() {
         if isCommandsEnabled, let typingCommand = typingCommand(in: composerView.inputMessageView.textView) {
             showCommandSuggestions(for: typingCommand)
             return
         }
+    }
 
+    open func updateMentionSuggestions() {
         if isMentionsEnabled, let (typingMention, mentionRange) = typingMention(in: composerView.inputMessageView.textView) {
             userMentionsDebouncer.execute { [weak self] in
                 self?.showMentionSuggestions(for: typingMention, mentionRange: mentionRange)
             }
             return
         }
+    }
 
+    open func updateSendMessageDisabled() {
         if !isSendMessageEnabled {
             composerView.inputMessageView.textView.placeholderLabel.text = L10n.Composer.Placeholder.messageDisabled
             composerView.recordButton.isHidden = true
             composerView.attachmentButton.isHidden = true
             composerView.commandsButton.isHidden = true
         }
         composerView.inputMessageView.isUserInteractionEnabled = isSendMessageEnabled
```

Comment:
Should we go for `updateSendMessage(_ isEnabled: Bool)` since currently it is always called and might be confusing if I override it?

Reply:
@laevandus can you re-review? 🙏 I've change the code a bit, and now a view visibility is only changed once